test/framework/deployments/testserver/deployment.go

```diff
@@ -114,6 +116,12 @@ func WithNodeSelector(selector map[string]string) DeploymentOptsFn {
 	}
 }
 
+func WithProtocol(protocol string) DeploymentOptsFn {
```

Comment:
Maybe `WithServicePortProtocol`?

Reply:
I renamed it to `WithServicePortAppProtocol`